src/components/Card.js

```diff
@@ -41,11 +51,25 @@ export default class Card {
     return this._cardElement;
   }
 
-  _handleCardDelete() {
-    const deleteCard = this._deletebutton.closest(".card");
-    deleteCard.remove();
-  }
   _handleCardLike() {
     this._likeButton.classList.toggle("card__like_liked");
+
+    if (this._likeCounter.textContent === "1") {
+      this._likeCounter.textContent = "0";
+      this._handleLike(this._data._id, "DELETE");
+    } else {
+      this._likeCounter.textContent = "1";

```

Comment:
You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.